docs: Improve generation of documentation

ThisIsMissEm opened the pull request:
Replays the latest docs generation changes from `solid-client-js` on this repo.

For comparison, the current docs: https://solid-client-access-grants-js.vercel.app/modules/type_IssueAccessRequestParameters.html

# Checklist

- [x] All acceptance criteria are met.
- [x] Relevant documentation, if any, has been written/updated.
- [ ] The changelog has been updated, if applicable.
- [ ] New functions/types have been exported in `index.ts`, if applicable.
- [ ] New modules (i.e. new `.ts` files) are listed in the `exports` field in `package.json`, if applicable.
- [ ] New modules (i.e. new `.ts` files) are listed in the `typedocOptions.entryPoints` field in `tsconfig.json`, if applicable.
- [ ] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).

ThisIsMissEm committed:
chore: remove vercel.json configuration

ThisIsMissEm committed:
docs: correctly document typescript interfaces

ThisIsMissEm committed:
chore: remove default commented config from tsconfig.json

ThisIsMissEm committed:
docs: improve documentation tooling

ThisIsMissEm committed:
docs: typescript interfaces and docs improvements

ThisIsMissEm commented:
@NSeydoux @matthieubosquet this isn't necessarily "done", but it would be good to get your eyes on it, as there are some code changes (some object types becoming concrete interfaces)

NSeydoux commented on docs/api/Makefile:
```diff
@@ -17,56 +17,45 @@ ifeq ($(MY_OS),Darwin)
-		$(error sphinx-build is not available, please follow the instructions in ./docs/api/README.md)
-	endif
+check:
+  $(if $(shell command -v $(SPHINXBUILD) 2> /dev/null),$(info Found `$(SPHINXBUILD)`),$(error sphinx-build is not available, please follow the instructions in ./docs/api/README.md))
```
```suggestion
  $(if $(shell command -v $(SPHINXBUILD) 2> /dev/null),$(info Found `$(SPHINXBUILD)`),$(error sphinx-build is not available, please follow the instructions in ./docs/README.md))
```

ThisIsMissEm replied:
Whoops! Good catch!

NSeydoux commented on docs/api/source/index.rst:
```diff
@@ -1,13 +1,102 @@
+====================================================
 
+``@inrupt/solid-client-access-grants`` is a JavaScript library for
+requesting and managing access given to an agent for a resource. These
+access grants are signed by a Solid Pod owner’s private key, which
+itself is managed by the Solid Pod server.
```
```suggestion
requesting and managing access given to an agent for a resource. These
access grants are represented by Verifiable Credentials, signed by an Issuer associated to the Pod server where the resources are hosted.
```
I'm afraid the notion of Pod owner could be confusing here: the private key is managed by the Pod server's operator, and not really associated to the resource owner.

ThisIsMissEm replied:
oh! Good call, we need to update the README.md too then!

NSeydoux commented on docs/api/source/index.rst:
```diff
@@ -1,13 +1,102 @@
+``Array.prototype.includes``, ``Object.entries`` and
+``String.prototype.endsWith``.
+
+Additionally, when using this package in an environment other than Node.js, you
+will need `a polyfill for Node's buffer module
+<https://www.npmjs.com/package/buffer>`__.
```
Is this the case for this library too, or is it only for solid-client-authn ?

ThisIsMissEm replied:
I'm not certain, tbh, so I'm kinda just going with "standardise this across projects" for now

ThisIsMissEm commented on src/type/AccessBaseOptions.ts:
```diff
@@ -19,13 +19,17 @@
 
+/**
+ * @module interfaces
+ */
```
An open question @NSeydoux & @matthieubosquet is: Do we want the interfaces to be located in `src/type/` or would we be better with them in `src/interfaces/` ?

NSeydoux replied:
I don't have a strong opinion, in fact I'm not even sure having them in a separate module is what we want. Do you have a preference between the two options ?

ThisIsMissEm committed:
docs: add sourcefile-url plugin

This makes the sourcefile URLs work regardless of environment or presence of the .git directory, meaning they work on Vercel and our hosted docs

ThisIsMissEm committed:
docs: Improve documentation of AccessBaseOptions

ThisIsMissEm committed:
refactor: use an explicit type for AccessModes

Instead of reusing solid-client-js's access.Access type, we instead have an explicit interface such that it can be documented along-side the Access Grants library

ThisIsMissEm commented on src/type/index.ts:
```diff
@@ -22,3 +22,4 @@
 export { FetchOptions } from "./FetchOptions";
 export { RedirectOptions } from "./RedirectOptions";
 export { AccessBaseOptions } from "./AccessBaseOptions";
+export { AccessModes } from "./AccessModes";
```
Aside: I think this should probably all be `export type` declarations instead maybe? I don't really know if there's that much of a difference though.

NSeydoux commented:
I went ahead and merged because the e2e failures are unrelated
